SourceEditorCompose/src/main/java/com/microsoft/device/display/samples/sourceeditorcompose/MainActivity.kt

```diff
@@ -0,0 +1,168 @@
+import com.microsoft.device.dualscreen.twopanelayout.TwoPaneLayout
+import com.microsoft.device.dualscreen.twopanelayout.TwoPaneScope
+import java.io.BufferedReader
+import java.io.InputStreamReader
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            TwoPaneExampleTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colors.background
+                ) {
+                    ControllerLayer()
+                }
+            }
+        }
+    }
+}
+
+/**
+ * PreviewPage Composable displays the preview for the HTML code present in the editor and updates it if there are any changes in the editor file
+ */
+@Composable
+fun TwoPaneScope.PreviewPage(text: String) {
+    val twoPaneScope = this
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    if (isSinglePane) {
+                        Text(text = stringResource(id = R.string.app_name))
+                    }
+                },
+                contentColor = Color.White,
+                backgroundColor = MaterialTheme.colors.primaryVariant,
+                actions = {
+                    if (twoPaneScope.isSinglePane) {
+                        IconButton(onClick = { twoPaneScope.navigateToPane1() }) {
+                            Text(text = stringResource(R.string.editor_button))
+                        }
+                    }
+                }
+            )
+        }
+    ) {
+        AndroidView(
+            factory = {
+                WebView(it).apply {
+                    layoutParams = ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT,
+                        ViewGroup.LayoutParams.MATCH_PARENT
+                    )
+                    webViewClient = WebViewClient()
+                    loadData(text, "text/html", "UTF-8")
+                }
+            },
+            update = {
+                it.loadData(text, "text/html", "UTF-8")
+            }
+        )
+    }
+}
+
+/**
+ * EditorPage Composable contains the Editor for the HTML file
+ */
+@Composable
+fun TwoPaneScope.EditorPage(text: String, updateText: (String) -> Unit) {
+    val twoPaneScope = this
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    Text(text = stringResource(R.string.app_name))
+                },
+                backgroundColor = MaterialTheme.colors.primaryVariant,
+                contentColor = Color.White,
+                actions = {
+                    if (twoPaneScope.isSinglePane) {
+                        IconButton(onClick = { twoPaneScope.navigateToPane2() }) {
+                            Text(text = stringResource(id = R.string.preview_button))
```

Comment:
I think to best follow material design guidelines, you can change the text buttons to icon buttons:
![image](https://user-images.githubusercontent.com/33138268/177438771-ae01a500-67cc-41b3-9538-07984261e6b8.png)

You can choose some icons from here: https://fonts.google.com/icons

Reply:
Added it!